modxcms/revolution
# Fix Dashboard Updates widget's display of the most recent modx version

### What does it do?
Added new `SoftwareUpdate` Processors for retrieving available MODX and Extras updates information, utilizing the new upgrades API. This PR is a Phase 1 fix that solves the issue at hand, but also paves the way for a more flexible and comprehensive Updates widget display to be done in a Phase 2 PR.

The new MODX download functionality provides a direct download instead of a link to the modx.com downloads area.

### Why is it needed?
The widget currently incorrectly shows the current -dev version.

### How to test
Clear browser and MODX caches and verify that the widget displays the correct update version. 
Test with an up-to-date `-pl` release (currently 3.0.5), a `-dev` release (github 3.0.x or 3.x), and an older `-pl` or `-dev` release (3.0.0 - 3.0.4).

### Example (with 3.0.4-dev installed)

https://github.com/user-attachments/assets/a74f8872-4047-4307-b1c0-0d522581066d



### Related issue(s)/PR(s)
Resolves #16466
